### PR TITLE
Storage configure fix

### PR DESCRIPTION
**Pull request by elorzafe**
*Issue #, if available:*
#2693 
*Description of changes:*
Update Storage.configure and Storage.vault.configure to support backwards compatibility (without using provider configuration) added unit test.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/__tests__/Storage-unit-test.ts [hunk 1]
```diff
@@ -49,28 +50,239 @@ describe('Storage', () => {
...
+
+            expect(config).toEqual({
+                AWSS3: {
+                    bucket: 'bucket',
+                    region: 'region'
+                }
+            });
+        });
+
+        test('Configure with Storage object', () => {
```
Comment (powerful23) [inline packages/storage/__tests__/Storage-unit-test.ts]: Do we really support something like: 
```js
storage.configure({
    Storage: {
    }
});
```

File: packages/storage/__tests__/Storage-unit-test.ts [hunk 2]
```diff
@@ -49,28 +50,239 @@ describe('Storage', () => {
...
+
+            expect(config).toEqual({
+                AWSS3: {
+                    bucket: 'bucket',
+                    region: 'region'
+                }
+            });
+        });
+
+        test('Configure with Storage and Provider object', () => {
```
Comment (powerful23) [inline packages/storage/__tests__/Storage-unit-test.ts]: Same question above.

File: packages/storage/__tests__/Storage-unit-test.ts (hunk 1 above)
Comment (elorzafe) [reply packages/storage/__tests__/Storage-unit-test.ts]: Not directly, but `Amplify.configure` will invoke it that way

Comment (powerful23) [reply packages/storage/__tests__/Storage-unit-test.ts]: Understood.

**Pull request merged**